presto-main/src/test/java/io/prestosql/tests/LogTestDurationListener.java

```diff
@@ -13,30 +13,125 @@
  */
 package io.prestosql.tests;
 
-import com.google.errorprone.annotations.concurrent.GuardedBy;
+import com.google.common.collect.ImmutableMap;
 import io.airlift.log.Logger;
 import io.airlift.units.Duration;
 import org.testng.IClassListener;
+import org.testng.IExecutionListener;
 import org.testng.IInvokedMethod;
 import org.testng.IInvokedMethodListener;
 import org.testng.ITestClass;
 import org.testng.ITestResult;
 
+import javax.annotation.concurrent.GuardedBy;
+
+import java.util.Arrays;
 import java.util.Map;
 import java.util.concurrent.ConcurrentHashMap;
-
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicLong;
+
+import static io.airlift.concurrent.Threads.daemonThreadsNamed;
 import static io.airlift.units.Duration.nanosSince;
 import static java.lang.String.format;
+import static java.util.stream.Collectors.joining;
 
 public class LogTestDurationListener
-        implements IClassListener, IInvokedMethodListener
+        implements IExecutionListener, IClassListener, IInvokedMethodListener
 {
     private static final Logger LOG = Logger.get(LogTestDurationListener.class);
 
     private static final Duration SINGLE_TEST_LOGGING_THRESHOLD = Duration.valueOf("30s");
     private static final Duration CLASS_LOGGING_THRESHOLD = Duration.valueOf("1m");
+    private static final Duration GLOBAL_IDLE_LOGGING_THRESHOLD = Duration.valueOf("2m");
```

Comment:
8m?
we just need to be below travis' 10m timeout

Reply:
I ran travis with 2m and only got the message for a real hang.  I'm thinking it might be nice when running in the IDE and you have a hung test, you would notice faster.

Reply:
My IDE reports current executing test(s).

Reply:
We surely have test classes taking more than 2m, but maybe we don't have as long test methods. Still I don't see any value 2m vs 8m. Up to you.

Reply:
I set it to 5m